src/sync/file-operations.ts

```diff
@@ -0,0 +1,74 @@
+import { format } from "util";
+import * as _ from "lodash";
+import * as fs from "fs";
+import * as path from "path";
+import * as zlib from "zlib";
+import { logger } from "../utils/logger";
+import { DB_WARP_SYNC_FILE_ENDING } from "../constants";
+
+export function getFileHash(filename: string): string {
+  const md5File = require("md5-file");
+  return md5File.sync(filename);
+}
+
+export async function compressAndHashFile(dbFileName: string, networkId: string, dbVersion: number, syncfileTemplate: string, directoryDir: string = ".") {
+  const WARP_SYNC_FILE = "__temp_sync_file__";
+  await createWarpSyncFile(directoryDir, dbFileName, WARP_SYNC_FILE);
+  const hash = getFileHash(path.join(directoryDir, WARP_SYNC_FILE));
+  const syncFile = format(syncfileTemplate, hash, networkId, dbVersion);
+  fs.renameSync(path.join(directoryDir, WARP_SYNC_FILE), path.join(directoryDir, syncFile));
+  logger.info(format("create warp sync file %s", syncFile));
+}
+
+export async function restoreWarpSyncFile(directoryDir: string, dbFileName: string, syncFilenameAbsPath: string) {
+  logger.info(format("restore/import warp sync file %s", syncFilenameAbsPath));
+  return new Promise<any>((resolve, reject) => {
+    const bigger = zlib.createGunzip();
+    const input = fs.createReadStream(syncFilenameAbsPath);
+    const output = fs.createWriteStream(path.join(directoryDir, dbFileName));
+
+    input
+      .pipe(bigger)
+      .pipe(output)
+      .on("error", (err: any) => {
+        logger.error("Error: restoring warp sync file");
+        reject(err);
+      })
+      .on("finish", () => {
+        resolve();
+      });
+  });
+}
+
+export async function createWarpSyncFile(directoryDir: string, dbFileName: string, syncFilename: string): Promise<any> {
+  return new Promise<any>((resolve, reject) => {
+    fs.closeSync(fs.openSync(path.join(directoryDir, syncFilename), "w"));
```

Comment:
"touch"ing the file? Is it necessary for the write stream? Would it be possible to do with promises insteqad of sync, since we're already in an sync function?

Reply:
removed touching file, not needed.